src/qz/communication/WinspoolEx.java

```diff
@@ -0,0 +1,25 @@
+package qz.communication;
+
+import com.sun.jna.Native;
+import com.sun.jna.Pointer;
+import com.sun.jna.platform.win32.WinNT;
+import com.sun.jna.platform.win32.Winspool;
+import com.sun.jna.win32.W32APIOptions;
+
+@SuppressWarnings("unused")
+public interface WinspoolEx extends Winspool {
+    WinspoolEx INSTANCE = Native.load("Winspool.drv", WinspoolEx.class, W32APIOptions.DEFAULT_OPTIONS);
```

Comment:
I am shocked this worked, we may want to make sure this is a good way to do this. I found no documentation about extending an interface from JNA.

Reply:
Whoops, I see `SetJob` is being added to the signature, so I take back that last part. 😅

Reply:
> I would expect this to work if the method signatures are similar enough however if there's any methods in Winspool that either aren't in WinspoolEx or take slightly different parameters in WinspoolEx, I would expect calling them to cause a crash.

Ok, this statement is off the mark.  Since you chose [`WinspoolEx`](https://stackoverflow.com/a/4039059/3196753) as a name, I assumed this actually existed.  What you're doing here is perfectly sane.  Pragmatically, we would open a PR to contribute these back to JNA contib.

Reply:
Yea i guess if they add `SetJob` to JNA this will break. It may make sense to pr JNA to add it, even though that will make that issue more likely. For now, I don't know of any other way to hook in `SetJob`

Reply:
Yes, but it would be a compilation error, and only if the signature was different, and only after we updated the dependency, so our CI would catch it. https://www.geeksforgeeks.org/the-override-annotation-in-java/.